### PR TITLE
Polkadot typegen

### PR DESCRIPTION
**Pull request by alessioscalici**
### Description

This pull request adds automatic Typescript type generation for Polkadot API to `nft-studio`

### Approvals

- [ ] Dev
- [ ] Dev

### CODE REVIEW

**Review by offerijns — APPROVED** [review]
Looks great to me! At some point we'll need to make it a bit more complex, as we'll have different metadata per runtime (and maybe per block if we want to handle migrations). But we don't need to needlessly overcomplicate this now.

File: centrifuge-js/src/utils/web3.ts [hunk 1]
```diff
@@ -1,13 +1,16 @@
 import { ApiPromise, WsProvider } from '@polkadot/api'
 import { RegistryTypes } from '@polkadot/types/types'
+import * as definitions from '../interfaces/definitions'
 
 const cached: { [key: string]: Promise<ApiPromise> } = {}
-export function getPolkadotApi(wsUrl: string, types?: RegistryTypes) {
+export function getPolkadotApi(wsUrl: string, regTypes?: RegistryTypes) {
+  const pdTypes = Object.values(definitions).reduce((res, { types }): object => ({ ...res, ...types }), {})
```
Comment (onnovisser) [inline centrifuge-js/src/utils/web3.ts:7]: These types should probably only get added when we're getting the parachain api. Also maybe an idea to return the cached api if available before doing this work.

Comment (alessioscalici) [reply centrifuge-js/src/utils/web3.ts:7]: Sorry, not sure I follow... aren't we doing this in this method already?

Comment (onnovisser) [reply centrifuge-js/src/utils/web3.ts:7]: not sure how big those definitions are but spreading the objects inside of a loop could be a bit slow. We could return `cached[wsUrl]` before the `reduce`

Comment (onnovisser) [reply centrifuge-js/src/utils/web3.ts:7]: but the main point was that this function gets called to get the parachain api, but also to get the relay chain api, which doesn't need the types

Comment (alessioscalici) [reply centrifuge-js/src/utils/web3.ts:7]: I will put the cache check and early return before the type loop, good point!

Would it make sense to split the 2 functions `getRelayChainApi` and `getParachainApi` at `centrifuge-js` level? WDYT?

**Comment by EstherCentrifuge** [comment]
This is deployed but "open" and 6+months old can you updaet please> @offerijns

**Comment by onnovisser** [comment]
Picking this up later

**Pull request closed without merging**